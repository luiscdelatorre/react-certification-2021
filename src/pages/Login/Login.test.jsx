import React from 'react';
import { fireEvent, getByTestId, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import AuthProvider from '../../providers/Auth';
import LoginPage from './Login.page';

describe('Login Page', () => {
  it('Check welcome message after render', () => {
    render(
      <AuthProvider>
        <LoginPage />
      </AuthProvider>
    );

    expect(screen.queryByText('Welcome back!')).toBeInTheDocument();
  });

  it('Should trigger login event', () => {
    const { container } = render(
      <BrowserRouter>
        <AuthProvider>
          <LoginPage />
        </AuthProvider>
      </BrowserRouter>
    );

    const button = getByTestId(container, 'btn-login');
    const event = fireEvent.click(button);

    expect(event).toBeTruthy();
  });

  it('Validates Layout snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <AuthProvider>
          <LoginPage />
        </AuthProvider>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
