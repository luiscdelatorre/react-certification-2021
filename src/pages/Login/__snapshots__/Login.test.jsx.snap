// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page Validates Layout snapshot 1`] = `
<section
  className="sc-bdfBwQ kQORl"
>
  <form
    className="sc-dlfnbm GCGce"
    onSubmit={[Function]}
  >
    <h1
      className="sc-gsTCUz dkhwEY"
    >
      Welcome back!
    </h1>
    <div
      className="sc-hKgILt gEPNix"
    >
      <label
        className="sc-eCssSg hBgXyk"
        htmlFor="username"
      >
        Username 
      </label>
      <input
        className="sc-jSgupP knAXbc"
        id="username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="sc-hKgILt gEPNix"
    >
      <label
        className="sc-eCssSg hBgXyk"
        htmlFor="password"
      >
        Password 
      </label>
      <input
        className="sc-jSgupP knAXbc"
        id="password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="sc-gKsewC hhyEnx"
      data-testid="btn-login"
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;
